org.eclipse.jdt.ls.core/src/org/eclipse/jdt/ls/core/internal/contentassist/SnippetCompletionProposal.java

```diff
@@ -311,6 +367,39 @@ private static CompletionItem getInterfaceSnippet(ICompilationUnit cu, Completio
 		return interfaceSnippetItem;
 	}
 
+	private static CompletionItem getRecordSnippet(SnippetCompletionContext scc, IProgressMonitor monitor) {
+		ICompilationUnit cu = scc.getCompilationUnit();
+		IJavaProject javaProject = cu.getJavaProject();
+		if (javaProject == null) {
+			return null;
+		}
+		AST ast = AST.newAST(javaProject.getOptions(true));
+		if (ast.apiLevel() < AST.JLS14) {//TODO is there a better way to check JDK version?
+			//not checking if preview features are enabled, as Java 14+ might support records without preview flag
+			return null;
+		}
```

Comment:
@snjeza lemme know if there's a better way to do this

Reply:
javaProject.getOption(JavaCore.COMPILER_COMPLIANCE, true)

Reply:
mmyeah. But it returns a String, that needs to be converted using internal JDT code to properly handle it